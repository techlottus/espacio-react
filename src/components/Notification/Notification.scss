@import '../../App.scss';

.noti-container {
  background-color: greenyellow;
  //position: fixed;
  //height: 1000vh;
  //width: 100vw;
  z-index: 19;
  //top: 0;
}

.notiitems {
  position: fixed;
  bottom: 0;
  width: calc(100% - 32px);
  //height: calc(100vh - 56px);
  z-index: 19;
  display: flex;
  flex-direction: column-reverse;
  margin: 0 pixeltorem(16px);
  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    width: calc(40% - 32px);
    margin: 0 pixeltorem(36px);
  }

  .notiitem {
    width: 100%;
    height: 56px;
    margin-bottom: pixeltorem(16px);
  }
  
}