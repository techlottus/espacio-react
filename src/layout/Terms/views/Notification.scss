@import '../../../App.scss';

.notification-container {
  align-items: center;
  display: flex;
   width: 100%;
  justify-content: space-between;
  padding: 1rem 0;
  border: 0.0625rem solid map-get($palette-neutral, n10);
  background-color: map-get($palette-neutral, n00);
  border-radius: 0.5rem;

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    width: calc(100% - 2rem);
  } 

  > .side {
    padding: 0 1rem;
    text-align: center;
    &.-normal {
      color: map-get($palette-neutral, n60);
    }
    > span {
      font-size: 1.5rem;
    }
  }

  > .content {
    > .content-text {
      width: 100%;
      min-height: 100%;
      font-size: pixeltorem(16px);
      &.isbold{
        font-weight: 700;
        font-size: pixeltorem(18px);
        line-height: pixeltorem(24px);
      }
    }
  }
}
