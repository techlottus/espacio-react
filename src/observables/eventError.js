import { Subject } from 'rxjs';

export const eventErrorAuth = new Subject();