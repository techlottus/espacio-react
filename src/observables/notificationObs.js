import { Subject } from 'rxjs'

export const notiObs = new Subject();

export const notiObsClose = new Subject();