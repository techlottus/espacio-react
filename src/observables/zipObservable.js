import { Subject } from 'rxjs'

export const zipObs = new Subject()