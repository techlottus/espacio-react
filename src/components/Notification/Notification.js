import React, { useEffect, useState } from "react";
import { notiObs, notiObsClose } from "../../observables/notificationObs";
import "./Notification.scss";
import { Feedback } from "../Feedback/Feedback";
import { feedbackNoti } from "../../constants/Notification.constant";

const Notification = () => {
  const [items, setItems] = useState([]);

  const handleItem = (data) => {
    return (
      <div className="notiitem" key={data.id}>
        <Feedback
          isNoti={true}
          data={{
            ...feedbackNoti,
            left: {
              name: data.icon,
              status: data.type,
            },
          }}
          text={data.text}
          html={() => {return}}
          onRight={() => {
            notiObsClose.next(data.id);
          }}
        />
      </div>
    );
  };

  const handleItems = () => {
    return (
      <div className="notiitems">
        {items.map((item) => {
          return handleItem(item);
        })}
      </div>
    );
  };

  useEffect(() => {
    notiObsClose.subscribe((id) => {
      setItems((items) => {
        return items.filter((item) => item.id !== id);
      });
    });

    notiObs.subscribe((info) => {
      let id = new Date().getTime();
      let timer = setTimeout(() => {
        setItems((items) => {
          return items.filter((item) => item.id !== id);
        });
        clearTimeout(timer);
      }, 60000);
      setItems((items) => {
        if (items.length === 3) {
          items.pop();
        }
        let newItems = [
          {
            icon: info.type.icon,
            type: info.type.type,
            text: info.text,
            id,
          },
          ...items,
        ];
        return newItems;
      });
    });
  }, []);

  return (
    <>
      <div className="noti-container">{handleItems()}</div>
    </>
  );
};

export default Notification;
