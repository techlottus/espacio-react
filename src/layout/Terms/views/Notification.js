import React from 'react'

import './Notification.scss'
const Notification = ({icon, text, isBold}) => {
  return (
    <>
      <div className='notification-container'>
        <div className="side -normal">
          <span className="material-icons icons">{icon}</span>
        </div>
        <div className="content">
          <p className={`content-text ${isBold ? 'isbold' : null }`}>{text}</p>
        </div>
        <div className="side"></div>
      </div>
    </>
  )
}

export default Notification;