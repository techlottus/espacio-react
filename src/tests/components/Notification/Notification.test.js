import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Notification from "../../../components/Notification/Notification";
import { notiObs } from "../../../observables/notificationObs";
import { Subject } from "rxjs";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState;

jest.doMock('../../../observables/notificationObs',() => {
  return {
    notiObs: new Subject(
      {
        type: {
          icon: 'person',
          type: 'success'
        },
        text: 'Hola Mundo'
      }
  )
  }
})


let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Notification  />
    </Provider>
  );
};

describe("Test <Notification/>", () => {

  beforeEach(() => {
    // notiObs.next({
    //   type: {
    //     icon: 'person',
    //     type: 'success'
    //   },
    //   text: 'Hola Mundo'
    // })
  })

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
