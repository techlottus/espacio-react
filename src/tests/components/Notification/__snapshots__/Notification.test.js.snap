// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Notification/> should init component 1`] = `
<div
  className="noti-container"
>
  <div
    className="notiitems"
  />
</div>
`;
